File: speeding_fine_tshirt/Graph/ScriptTemplate/LessThan/CGLessThan.lua
local CGLessThan = CGLessThan or {}
CGLessThan.__index = CGLessThan

function CGLessThan.new()
    local self = setmetatable({}, CGLessThan)
    self.inputs = {}
    return self
end

function CGLessThan:setInput(index, func)
    self.inputs[index] = func
end

function CGLessThan:getOutput(index)
    return self.inputs[0]() < self.inputs[1]()
end

return CGLessThan

File: speeding_fine_tshirt/Graph/ScriptTemplate/LessThan/CGLessThan.js
/**
 * @file CGLessThan.js
 * @author
 * @date 2021/8/15
 * @brief CGLessThan.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGLessThan extends BaseNode {
  constructor() {
    super();
  }

  getOutput() {
    return this.inputs[0]() < this.inputs[1]();
  }
}

exports.CGLessThan = CGLessThan;
